### PR TITLE
docs: add sample gallery contributing guide

### PR DESCRIPTION
**Pull request by HuihuiWu-Microsoft**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -9,6 +10,53 @@ When you submit a pull request, a CLA-bot will automatically determine whether y
...
 
+## How to submit your sample
+To give you an overview of the process to contribute to our samples repo, you will be going through this:
+
+![Sample Onboarding Process](../assets/contributing_process.png)
+
+> The purple blocks are things you will need to do.
+> The green blocks are what we will take care of.
+
+Before your start, you might want to know whether you will need to commit your source code to our repo. In most of cases, your Teams sample source code will need to be commited to our repo, especially when you are an individual contributor. In some special cases if you are our partner team who maintains Teams samples repo as well, your source code can be resides in your own open source repo. We will reference to your repo and expose your sample to Teams Toolkit sample gallery.
```
Comment (tecton) [inline CONTRIBUTING.md:21]: "your source code can be resides in your own open source repo": can reside in / can live in

Comment (tecton) [reply CONTRIBUTING.md:21]: "We will reference to": We will refer to?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -9,6 +10,53 @@ When you submit a pull request, a CLA-bot will automatically determine whether y
...
+
+Some explaination of how to perform each step in above process:
+1. Submit your source code by creating a pull request to our repo (target `dev` branch by default), a typical project is in below structure. Make sure you have the `sampleDemo.gif` and `README.md` files included in your source code.
```
Comment (tecton) [inline CONTRIBUTING.md:24]: `sampleDemo.gif` should be changed to `thumbnail.png` now.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -9,6 +10,53 @@ When you submit a pull request, a CLA-bot will automatically determine whether y
...
+```
+- sampleTeamsProject
+  - appPackage 
+      - color.png
+      - manifest.json
+      - outline.png 
+  - env 
+  - src 
+  - assets
+      - sampleDemo.gif
```
Comment (tecton) [inline CONTRIBUTING.md:37]: Same here.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -9,6 +10,53 @@ When you submit a pull request, a CLA-bot will automatically determine whether y
...
+  - README.md
+  - teamsapp.yml 
+  - teamsapp.local.yml
+```
+The sampleDemo.gif is a very short gif illustrating what users will see after the app is running, and demonstrating major usage of the sample. The gif can also be a static screenshot of the sample if you want.
```
Comment (tecton) [inline CONTRIBUTING.md:42]: Shall we add a description of thumbnail too?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -9,6 +10,53 @@ When you submit a pull request, a CLA-bot will automatically determine whether y
...
+> Suggested size of gif/pic is *1600x920/800x460* (If you have it with other sizes, please keep same image ratio for other sizes and don't forget to consider the resolution. Your gif or pics must fit the entire Teams client UI into the image.
+
+The README.md file basically describe what's the sample, how to run it and so on. Here is a [README Template](https://github.com/OfficeDev/TeamsFx-Samples/blob/zhany-readme-template/README_template.md) which you can follow to create one.
```
Comment (tecton) [inline CONTRIBUTING.md:45]: Can we also include README template in this PR?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -9,6 +10,53 @@ When you submit a pull request, a CLA-bot will automatically determine whether y
...
+
+2. [Log a Issue ticket](https://github.com/OfficeDev/TeamsFx-Samples/issues/new?assignees=summzhan&labels=&projects=&template=new-sample-onboard-request.md&title=%5BSample+Onboard+Request%5D+Your+sample+name) in our repo using below template, you will need to provide meta data of your sample in your ticket.
```
Comment (tecton) [inline CONTRIBUTING.md:47]: Log a Issue ticket -> Log an Issue Ticket

**Review by tecton — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (HuihuiWu-Microsoft) [reply CONTRIBUTING.md:45]: I think it might be too large. Add @summzhan here to confirm

File: CONTRIBUTING.md (hunk 6 above)
Comment (HuihuiWu-Microsoft) [reply CONTRIBUTING.md:47]: I'd like to change it to 'Log an issue ticket' to align with other parts

File: CONTRIBUTING.md (hunk 1 above)
Comment (HuihuiWu-Microsoft) [reply CONTRIBUTING.md]: Updated

**Review by summzhan — DISMISSED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (summzhan) [reply CONTRIBUTING.md:46]: link is enough. I approve this PR

Comment (HuihuiWu-Microsoft) [reply CONTRIBUTING.md]: Updated

**Pull request merged**